### ISSUE TITLE
Question about angleproto.py code

### ISSUE THREAD

**Issue by huangrunqian** [clovaai/voxceleb_trainer#146]
Why do you use fake labels instead of the input real labels here?
---> from angleproto.py

    def forward(self, x, label=None):

        assert x.size()[1] >= 2

        out_anchor      = torch.mean(x[:,1:,:],1)
        out_positive    = x[:,0,:]
        stepsize        = out_anchor.size()[0]

        cos_sim_matrix  = F.cosine_similarity(out_positive.unsqueeze(-1),out_anchor.unsqueeze(-1).transpose(0,2))
        torch.clamp(self.w, 1e-6)
        cos_sim_matrix = cos_sim_matrix * self.w + self.b
        
        label   = torch.from_numpy(numpy.asarray(range(0,stepsize))).cuda()
        nloss   = self.criterion(cos_sim_matrix, label)
        prec1   = accuracy(cos_sim_matrix.detach(), label.detach(), topk=(1,))[0]

        return nloss, prec1

**Comment by huangrunqian** [comment]
Made it out finally~

**Comment by huangrunqian** [comment]
I just use the loss function, while the model and data are different, so the result is not comparable. Maybe you could check and compare your result with that of the original paper. As you use the speaker labels, it's a supervised learning method.
